src/other/utils.jl

```diff
@@ -54,6 +54,22 @@ julia> select(df1, AsTable([:a, :b]) => ByRow(nt -> map(x -> x^2, nt)) => AsTabl
 """
 struct AsTable
     cols
+
+    # the rules used here are simplified as not all column selectors
+    # that pass through them are valid, but this is safe enough in practice
+    function AsTable(cols)
+        if !(cols isa Union{AbstractString, Symbol, Signed, Unsigned,
+                          AbstractVector{<:Integer},
+                          AbstractVector{Symbol},
+                          AbstractVector{<:AbstractString},
+                          Colon, Not, Between, All, Cols, Regex})
+            if !(cols isa AbstractVector &&
+                 all(col -> col isa Union{Integer, Symbol, AbstractString}, cols))
+                throw(ArgumentError("Unrecognized column selector $cols"))
+            end
+        end
+        return new(cols)
```

Comment:
I find this a bit simpler to grasp:
```suggestion
        if cols isa Union{AbstractString, Symbol, Signed, Unsigned,
                          AbstractVector{<:Integer},
                          AbstractVector{Symbol},
                          AbstractVector{<:AbstractString},
                          Colon, Not, Between, All, Cols, Regex}) ||
            (cols isa AbstractVector &&
             all(col -> col isa Union{Integer, Symbol, AbstractString}, cols))
                return new(cols)
            else
                throw(ArgumentError("Unrecognized column selector $cols"))
            end
        end
```

Reply:
OK - I thought you would find it less readable :).